__init__.py

```diff
@@ -106,40 +113,93 @@ def respond(self, query):
         if len(query) == 0:
             return 0.0
 
-        r = ddg.query(query)
+        # note: '1+1' throws an exception
+        try:
+            r = ddg.query(query)
+        except:
+            LOG.warning("DDG exception!")
+            return None
 
-        LOG.debug('Query: ' + str(query))
-        LOG.debug('Type: ' + r.type)
+        LOG.debug("Query: %s" % (str(query),))
+        LOG.debug("Type: %s" % (r.type,))
+
+        # if disambiguation, save old result
+        # for fallback but try to get the 
+        # real abstract
+        if r.type == 'disambiguation':
+            if r.related:
+                detailed_url = r.related[0].url + "?o=x"
+                LOG.debug("DDG: disambiguating %s" % (detailed_url,))
+                request = requests.get(detailed_url)
+                response = request.content
+                if response:
+                    xml = ElementTree.fromstring(response)
+                    r = ddg.Results(xml)
 
         if (r.answer is not None and r.answer.text and
                 "HASH" not in r.answer.text):
             return(query + self.is_verb + r.answer.text + '.')
         elif len(r.abstract.text) > 0:
             sents = split_sentences(r.abstract.text)
-            return sents[0]
+            #return sents[0]  # what it is
+            #return sents     # what it should be
+            return ". ".join(sents)   # what works for now
         elif len(r.related) > 0 and len(r.related[0].text) > 0:
             related = split_sentences(r.related[0].text)[0]
             return(self.format_related(related, query))
         else:
             return None
 
+    def fix_input(self, query):
+        for noun in self.translated_question_words:
+            for verb in self.translated_question_verbs:
+                for article in [i + ' ' for i in self.translated_articles] + ['']:
+                    test = noun + verb + ' ' + article
```

Comment:
why no space between noun and verb?  I noticed this was in the wiki skill too.  Should it be added to the common query class?

Reply:
I don't know exactly why no space. This code predates me but I will cut you a ticket for further investigation. I do not think this should be added to the CommonQuery class as it is an ugly way to go about one's business.